### PR TITLE
Add hardhat networks settings external tests

### PR DESCRIPTION
**Pull request by r0qs**
Note that `t_ems_ext_hardhat` will be fixed by PR https://github.com/ethereum/solidity/pull/14933 when we switch evm version to cancun by default.

### CODE REVIEW

File: test/externalTests/elementfi.sh [hunk 1]
```diff
@@ -107,6 +107,7 @@ function elementfi_test
     force_hardhat_compiler_settings "$config_file" "$(first_word "$SELECTED_PRESETS")" "$config_var"
     force_hardhat_unlimited_contract_size "$config_file" "$config_var"
     npm install
+    npm install hardhat@2.20.0
```
Comment (cameel) [inline test/externalTests/elementfi.sh]: Is it really a version bump? Looks like downgrade.

Also, missing a comment saying why this specific version and what has to happen for us to be able to remove this.

**Review by cameel — DISMISSED** [review]
Seems to work, but you should add a comment about the version bump (downgrade)?

File: test/externalTests/elementfi.sh (hunk 1 above)
Comment (r0qs) [reply test/externalTests/elementfi.sh]: Yes, indeed, I will adjust the commit message. Also, all external tests are currently utilizing the `<latest-release>` as ref. Thus, we may consider reintroducing the `neutralize_package_lock` and removing the need to install a specific version of hardhat. Maybe that is better.

Comment (r0qs) [reply test/externalTests/elementfi.sh]: Hum...apparently this does not work for `pool_together`: https://app.circleci.com/pipelines/github/ethereum/solidity/33310/workflows/1f695a1b-e2d1-4f4b-8ac6-5da65e628182/jobs/1498929 but fix the `hardhat` version to `2.20.0` does. So I will rollback that change for this external test.

**Review by cameel — APPROVED** [review]

**Pull request merged**